src/rpc/dial.rs

```diff
@@ -329,13 +329,8 @@ impl<T: AuthMethod> DialBuilder<T> {
         let mut resp: Option<Response> = None;
         for ipv4 in addresses {
             for candidate in candidates {
-                let mut addr_to_send = "".to_string();
-                addr_to_send.push_str(candidate.as_str());
-                addr_to_send.push('.');
-                addr_to_send.push_str(VIAM_MDNS_SERVICE_NAME);
-
                 let discovery = discover::interface_with_loopback(
-                    addr_to_send,
+                    VIAM_MDNS_SERVICE_NAME,
```

Comment:
See added comment re: the localness of machine URIs to understand why this is important. Basically, we were trying to discover candidates with a fully qualified name. but, the fully qualified name we were passing was non-local and, for reasons not entirely clear to me, the only candidates we could find from a micro-rdk process were the ones with a local name. By making our search more generic we can find everything that looks Viam-y, and then narrow it down based on the actual name of the machine.